daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala

```diff
@@ -138,24 +138,25 @@ private[lf] object Pretty {
         text("due to a fetch of a consumed contract") & prettyContractId(coid) &
           char('(') + (prettyIdentifier(tid)) + text(").") /
             text("The contract had been consumed in transaction") & prettyEventId(consumedBy)
-      case ScenarioErrorContractNotVisible(coid, tid, committer, observers) =>
+      case ScenarioErrorContractNotVisible(coid, tid, actAs, readAs, observers) =>
         text("due to the failure to fetch the contract") & prettyContractId(coid) &
           char('(') + (prettyIdentifier(tid)) + text(").") /
-            text("The contract had not been disclosed to the committer") & prettyParty(committer) + char(
+            text(
+              s"The contract had not been disclosed to the reading parties: actAs: ${actAs}, readAs: ${readAs}") & char(
           '.',
         ) /
           text("The contract had been disclosed to:") & intercalate(
           comma + space,
           observers.map(prettyParty),
         ) + char('.')
-      case ScenarioErrorContractKeyNotVisible(coid, gk, committer, stakeholders) =>
+      case ScenarioErrorContractKeyNotVisible(coid, gk, actAs, readAs, stakeholders) =>
         text("due to the failure to fetch the contract") & prettyContractId(coid) &
           char('(') + (prettyIdentifier(gk.templateId)) + text(") associated with key ") +
             prettyValue(false)(gk.key) &
-          text("The contract had not been disclosed to the committer") & prettyParty(committer) + char(
-          '.',
-        ) /
-          text("Stakeholders:") & intercalate(
+          text(
+            s"The contract had not been disclosed to the reading parties: actAs ${actAs}, readAs: ${readAs}") + char(
+            '.') /
```

Comment:
I have a mixed feelings about stopping using the pretty print construct from `org.typelevel.paiges.Doc`. I am not  sure it was very usefull at the first place, but I do not like the state of the file now.